### PR TITLE
Fix init container resources reference error

### PR DESCRIPTION
**Pull request by kebe7jun**
- [X] Installation

### CODE REVIEW

File: manifests/charts/istio-control/istio-discovery/files/injection-template.yaml [hunk 1]
```diff
@@ -110,7 +110,7 @@ spec:
     {{- end }}
   {{- else }}
     {{- if .Values.global.proxy.resources }}
```
Comment (bianpengyuan) [inline manifests/charts/istio-control/istio-discovery/files/injection-template.yaml:112]: This should also be changed to proxy_init.

Comment (kebe7jun) [reply manifests/charts/istio-control/istio-discovery/files/injection-template.yaml]: Thanks! : )

**Review by howardjohn — CHANGES_REQUESTED** [review]
I don't think this is right. It looks right at first glance, but:

* It is a breaking change
* There is no reason I know of to have a different request for init vs container; if we request 1 CPU in the sidecar then its fine to request 1 CPU in the init container as well, as it will not have any negative impact on scheduling
* The request must be higher. IIRC the reason we made the change ^ is specifically because the 10m CPU requests led to the container getting throttled and taking >10s to setup iptables.
* Annotations like `sidecar.istio.io/proxyCPU` already apply to both.

See https://github.com/istio/istio/issues/26932

I would prefer if we just warn if its set maybe

**Comment by kebe7jun** [comment]
> I don't think this is right. It looks right at first glance, but:
> 
> * It is a breaking change
> * There is no reason I know of to have a different request for init vs container; if we request 1 CPU in the sidecar then its fine to request 1 CPU in the init container as well, as it will not have any negative impact on scheduling
> * The request must be higher. IIRC the reason we made the change ^ is specifically because the 10m CPU requests led to the container getting throttled and taking >10s to setup iptables.
> * Annotations like `sidecar.istio.io/proxyCPU` already apply to both.
> 
> See #26932
> 
> I would prefer if we just warn if its set maybe

Right, so should we remove the `global.proxy_init.resources` setting to avoid misunderstanding it?

**Pull request closed without merging**